src/Stripe.net/Infrastructure/Public/SystemNetHttpClient.cs

```diff
@@ -60,24 +67,62 @@ public static System.Net.Http.HttpClient BuildDefaultSystemNetHttpClient()
             StripeRequest request,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            var httpRequest = BuildRequestMessage(request);
+            TimeSpan duration;
+            HttpResponseMessage response = null;
+            int retry = 0;
+            OperationCanceledException timeoutException = null;
+
+            this.requestTelemetry.MaybeAddTelemetryHeader(request.StripeHeaders);
+
+            while (true)
+            {
+                timeoutException = null;
+
+                var httpRequest = BuildRequestMessage(request);
 
-            this.requestTelemetry.MaybeAddTelemetryHeader(httpRequest.Headers);
+                var stopwatch = Stopwatch.StartNew();
 
-            var stopwatch = Stopwatch.StartNew();
-            var response = await this.httpClient.SendAsync(httpRequest, cancellationToken)
-                .ConfigureAwait(false);
-            stopwatch.Stop();
+                try
+                {
+                    response = await this.httpClient.SendAsync(httpRequest, cancellationToken)
+                        .ConfigureAwait(false);
+                }
+                catch (OperationCanceledException exception)
+                    when (!cancellationToken.IsCancellationRequested)
```

Comment:
Internally, `HttpClient` uses a `CancelationToken` to handle timeouts, so we need to use a `when` filter to distinguish between timeouts and user-requested cancelations.